dashboards-observability/public/components/visualizations/charts/stats/stats.tsx

```diff
@@ -0,0 +1,441 @@
+  };
+
+  const ZERO_ERROR_ANNOTATION = 0.01;
+  let autoChartLayout: any = {
+    xaxis: {
+      visible: false,
+      showgrid: false,
+      anchor: 'y1',
+    },
+    yaxis: {
+      visible: false,
+      showgrid: false,
+      anchor: 'x1',
+    },
+    annotations: [],
+  };
+
+  const selectedDimensionsData = dimensions
+    .map((dimension: any) => data[dimension.name])
+    .reduce((prev, cur) =>
+      prev.map((item: string | number, index: number) => `${item},<br>${cur[index]}`)
+    );
+
+  const createValueText = (value: string | number) =>
+    `<b>${value}${
+      metricUnits ? `<span style="font-size: ${metricUnitsSize}px"}> ${metricUnits}</span>` : ''
+    }</b>`;
+
+  const createAnnotationsAutoChartHorizontal = ({
+    label,
+    value,
+    index,
+    valueColor,
+  }: createAnnotationType) =>
+    textMode === 'values+names' || textMode === DefaultTextMode
+      ? [
+          {
+            ...STATS_ANNOTATION,
+            x: 0 + ZERO_ERROR_ANNOTATION,
+            y:
+              index > 0
+                ? (index + 1) / metricsLength - ZERO_ERROR_ANNOTATION
+                : 1 / metricsLength - ZERO_ERROR_ANNOTATION,
+            xanchor: 'left',
+            yanchor: 'bottom',
+            text: label,
+            font: {
+              size: titleSize,
+              color: isDarkMode ? COLOR_WHITE : COLOR_BLACK,
+              family: 'Roboto',
+            },
+            type: 'name',
+            metricValue: value,
+          },
+          {
+            ...STATS_ANNOTATION,
+            x: 1,
+            y:
+              index > 0
+                ? (index + 1) / metricsLength - ZERO_ERROR_ANNOTATION
+                : 1 / metricsLength - ZERO_ERROR_ANNOTATION,
+            xanchor: 'right',
+            yanchor: 'bottom',
+            text: createValueText(value),
+            font: {
+              size: valueSize,
+              color: valueColor,
+              family: 'Roboto',
+            },
+            type: 'value',
+            metricValue: value,
+          },
+        ]
+      : [
+          {
+            ...STATS_ANNOTATION,
+            x: 0.5,
+            y:
+              metricsLength === 1
+                ? 0.5
+                : index === 0
+                ? ((1 / metricsLength) * 1) / 2
+                : (index + 1) / metricsLength - ((1 / metricsLength) * 1) / 2,
+            xanchor: 'center',
+            yanchor: 'bottom',
+            text: textMode === 'values' ? createValueText(value) : label,
+            font: {
+              size: textMode === 'values' ? valueSize : titleSize,
+              color: textMode === 'names' ? (isDarkMode ? COLOR_WHITE : COLOR_BLACK) : valueColor,
+              family: 'Roboto',
+            },
+            type: textMode === 'names' ? 'name' : 'value',
+            metricValue: value,
+          },
+        ];
+
+  const createAnnotation = ({ label, value, index, valueColor }: createAnnotationType) =>
```

Comment:
can we combine these two methods `createAnnotation ` and `createAnnotationsAutoChartHorizontal `?

Reply:
both orientations have different x, y values and calculations